dra7xx.mk

```diff
@@ -0,0 +1,145 @@
+	$(LINUX_PATH)/ti_config_fragments/baseport.cfg \
+	$(LINUX_PATH)/ti_config_fragments/ipc.cfg \
+	$(LINUX_PATH)/ti_config_fragments/connectivity.cfg \
+	$(LINUX_PATH)/ti_config_fragments/audio_display.cfg \
+	$(LINUX_PATH)/ti_config_fragments/wlan.cfg \
+	$(LINUX_PATH)/ti_config_fragments/omap_soc.cfg \
+	$(LINUX_PATH)/ti_config_fragments/lpae.cfg \
+	$(LINUX_PATH)/ti_config_fragments/dra7_only.cfg \
+	$(LINUX_PATH)/ti_config_fragments/debug_options.cfg
+
+linux-defconfig: $(LINUX_PATH)/.config
+
+LINUX_COMMON_FLAGS += ARCH=arm
+linux: linux-common
+linux-defconfig-clean: linux-defconfig-clean-common
+
+LINUX_CLEAN_COMMON_FLAGS += ARCH=arm
+linux-clean: linux-clean-common
+
+LINUX_CLEANER_COMMON_FLAGS += ARCH=arm
+linux-cleaner: linux-cleaner-common
+
+###############################################################################
+# OP-TEE
+###############################################################################
+.PHONY: optee-os optee-os-clean optee-client optee-client-clean
+
+OPTEE_OS_COMMON_FLAGS += PLATFORM=ti-dra7xx
+optee-os: optee-os-common
+
+OPTEE_OS_CLEAN_COMMON_FLAGS += PLATFORM=ti-dra7xx
+optee-os-clean: optee-os-clean-common
+
+optee-client: optee-client-common
+optee-client-clean: optee-client-clean-common
+
+###############################################################################
+# xtest / optee_test
+###############################################################################
+.PHONY: xtest xtest-clean xtest-patch
+
+xtest: xtest-common
+xtest-clean: xtest-clean-common
+xtest-patch: xtest-patch-common
+
+###############################################################################
+# hello_world
+###############################################################################
+.PHONY: helloworld helloworld-clean
+
+helloworld: helloworld-common
+helloworld-clean: helloworld-clean-common
+
+###############################################################################
+# Busybox
+###############################################################################
+.PHONY: busybox busybox-clean busybox-cleaner
+
+BUSYBOX_COMMON_TARGET = dra7xx
+BUSYBOX_CLEAN_COMMON_TARGET = dra7xx clean
+
+busybox: busybox-common
+busybox-clean: busybox-clean-common
+busybox-cleaner: busybox-cleaner-common
+
+###############################################################################
+# Root FS
+###############################################################################
+.PHONY: filelist-tee update_rootfs
+
+filelist-tee: filelist-tee-common
+update_rootfs: update_rootfs-common
+
+###############################################################################
+# Build FIT
+###############################################################################
+.PHONY: build-fit build-fit-clean
+
+build-fit: linux update_rootfs optee-os
+	cp $(LINUX_IMAGE) $(STAGEING_AREA)/
+	cp $(LINUX_DTBS) $(STAGEING_AREA)/
+	cp $(OPTEE_OS_BIN) $(STAGEING_AREA)/
+	MKIMAGE=$(U-BOOT_PATH)/tools/mkimage $(MAKE) -C $(STAGEING_AREA)
```

Comment:
This will create `fitImage.itb`. It would be quite nice to have that populated automatically to the /boot/ folder in the rootfs (i.e, `gen_rootfs/filesystem.cpio.gz`). Not that long ago each Makefile in build.git had separate sections where you could add specific files. Now that has been placed in the common.mk file instead (target `filelist-tee-common`). I have no straight answer how that could be added after we've changed to the common file. I can see that hikey.mk still has some code, but to me it looks like most of that code shouldn't be there any longer.